src/BloomBrowserUI/bookEdit/js/origami.js

```diff
@@ -65,31 +76,79 @@ function layoutToggleClickHandler() {
 function splitClickHandler() {
     var myInner = $(this).closest('.split-pane-component-inner');
     if ($(this).hasClass('splitter-top'))
-        performSplit2(myInner, 'horizontal', 'bottom', 'top');
+        performSplit(myInner, 'horizontal', 'bottom', 'top', true);
     else if ($(this).hasClass('splitter-right'))
-        performSplit(myInner, 'vertical', 'left', 'right');
+        performSplit(myInner, 'vertical', 'left', 'right', false);
     else if ($(this).hasClass('splitter-bottom'))
-        performSplit(myInner, 'horizontal', 'top', 'bottom');
+        performSplit(myInner, 'horizontal', 'top', 'bottom', false);
     else if ($(this).hasClass('splitter-left'))
-        performSplit2(myInner, 'vertical', 'right', 'left');
+        performSplit(myInner, 'vertical', 'right', 'left', true);
 }
 
-function performSplit(innerElement, verticalOrHorizontal, existingContentPosition, newContentPosition) {
+function performSplit(innerElement, verticalOrHorizontal, existingContentPosition, newContentPosition, prependNew) {
+    addUndoItem(innerElement.parent());
     innerElement.wrap(getSplitPaneHtml(verticalOrHorizontal));
     innerElement.wrap(getSplitPaneComponentHtml(existingContentPosition));
     var newSplitPane = innerElement.closest('.split-pane');
-    newSplitPane.append(getSplitPaneDividerHtml(verticalOrHorizontal));
-    newSplitPane.append(getSplitPaneComponentWithNewContent(newContentPosition));
+    if (prependNew) {
+        newSplitPane.prepend(getSplitPaneDividerHtml(verticalOrHorizontal));
+        newSplitPane.prepend(getSplitPaneComponentWithNewContent(newContentPosition));
+    } else {
+        newSplitPane.append(getSplitPaneDividerHtml(verticalOrHorizontal));
+        newSplitPane.append(getSplitPaneComponentWithNewContent(newContentPosition));
+    }
     newSplitPane.splitPane();
 }
 
-function performSplit2(innerElement, verticalOrHorizontal, existingContentPosition, newContentPosition) {
-    innerElement.wrap(getSplitPaneHtml(verticalOrHorizontal));
-    innerElement.wrap(getSplitPaneComponentHtml(existingContentPosition));
-    var newSplitPane = innerElement.closest('.split-pane');
-    newSplitPane.prepend(getSplitPaneDividerHtml(verticalOrHorizontal));
-    newSplitPane.prepend(getSplitPaneComponentWithNewContent(newContentPosition));
-    newSplitPane.splitPane();
+var origamiUndoStack = [];
+var origamiUndoIndex = 0; // of item that should be redone next, if any
+
+// I made this take an argument of the closest parent that survies the change with only
+// its children modified. I could not make it work for multiple levels of Undo and Redo
+// (including deletions) by only cloning that element, as I first attempted; but I
+// decided to keep the argument in case something requires us to attempt that again one day.
+function addUndoItem(parentElement) {
+    origamiUndoStack.length = origamiUndoIndex; // truncate any redo items
```

Comment:
So in light of what's it's doing... is a better name "addUndoPoint"  (as in "point in time")?